pkg/ctlog/config_test.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	if configIn.PrivKey == nil || configOut.PrivKey == nil || !configOut.PrivKey.Equal(configIn.PrivKey) {
+		t.Errorf("Private Keys differ")
+	}
+}
+
+func TestAddNewFulcioAndRemoveOld(t *testing.T) {
+	ctx := context.TODO()
+	in, err := createBaseConfig(t)
+	if err != nil {
+		t.Fatalf("failed to createBaseConfig: %v", err)
+	}
+	config, err := Unmarshal(ctx, in)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal: %v", err)
+	}
+
+	newFulcioCert, err := createTestCert(t)
+	if err != nil {
+		t.Fatalf("Failed to create a test certificate: %v", err)
+	}
+	config.AddFulcioRoot(ctx, newFulcioCert)
+	marshaled, err := config.MarshalConfig(context.Background())
+	if err != nil {
+		t.Fatalf("Failed to MarshalConfig: %v", err)
+	}
+
+	// Now test that we have configuration that trusts both Fulcio roots
+	// simulating while one is being spun down.
+	expected := [][]byte{}
+	expected = append(expected, []byte(existingRootCert), newFulcioCert)
+	validateFulcioEntries(ctx, marshaled, expected, t)
+
+	newConfig, err := Unmarshal(ctx, marshaled)
+	if len(newConfig.FulcioCerts) != 2 {
+		t.Fatalf("Unexpected number of FulcioCerts, got %d", len(newConfig.FulcioCerts))
+	}
+
+	// Now for our next trick, pretend we're rotating, so take out the
+	// existing entry from the trusted certs.
+	newConfig.RemoveFulcioRoot(ctx, []byte(existingRootCert))
+	marshaledNew, err := newConfig.MarshalConfig(context.Background())
+	if err != nil {
+		t.Fatalf("Failed to marshal new configuration after removal: %v", err)
+	}
+
+	// Now test that we have configuration that trusts only the new Fulcio
+	// root, simulating that the old one has been spun down.
+	expected = make([][]byte, 0)
+	expected = append(expected, []byte(newFulcioCert))
+	validateFulcioEntries(ctx, marshaledNew, expected, t)
+}
+
+func createBaseConfig(t *testing.T) (map[string][]byte, error) {
+	t.Helper()
+	c, err := b64.StdEncoding.DecodeString(testConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to decode testConfig: %w", err)
+	}
+	private, err := b64.StdEncoding.DecodeString(privateKeyEncoded)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to decode privateKeyEncoded: %w", err)
+	}
+	public, err := b64.StdEncoding.DecodeString(publicKeyEncoded)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to decode publicKeyEncoded: %w", err)
+	}
+	return map[string][]byte{
+		"config":  c,
+		"private": private,
+		"public":  public,
+		"rootca":  []byte(existingRootCert),
+	}, nil
+}
+
+func createTestCert(t *testing.T) ([]byte, error) {
+	// Generate RSA key.
+	key, err := rsa.GenerateKey(rand.Reader, bitSize)
+	if err != nil {
+		return nil, fmt.Errorf("GenerateKey failed: %w", err)
+	}
+	// Extract public component.
+	pub := key.Public()
+
+	serialNumber, err := rand.Int(rand.Reader, new(big.Int).SetInt64(math.MaxInt64))
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate serial Number: %w", err)
+	}
+	rootCA := &x509.Certificate{
```

Comment:
See https://github.com/sigstore/rekor/blob/main/pkg/pki/x509/testutils/cert_test_utils.go, I created some test utils for creating root certs that are more representative of Sigstore root certs.